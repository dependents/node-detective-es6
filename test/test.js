var detective = require("../");

describe("detective-es6", function () {
  var ast = {
    type: "Program",
    body: [
      {
        type: "VariableDeclaration",
        declarations: [
          {
            type: "VariableDeclarator",
            id: {
              type: "Identifier",
              name: "x",
            },
            init: {
              type: "Literal",
              value: 4,
              raw: "4",
            },
          },
        ],
        kind: "let",
      },
    ],
  };

  it("accepts an ast", function () {
    var deps = detective(ast);
    expect(deps).toMatchSnapshot();
  });

  it("retrieves the dependencies of es6 modules", function () {
    var deps = detective('import Abc, * as All from "mylib";');
    expect(deps).toMatchSnapshot();
  });

  it("retrieves the re-export dependencies of es6 modules", function () {
    var deps = detective('export {foo, bar} from "mylib";');
    expect(deps).toMatchSnapshot();
  });

  it("retrieves the re-export dependencies alias of es6 modules", function () {
    var deps = detective('export {foo as Foo} from "mylib";');
    expect(deps).toMatchSnapshot();
  });

  it("retrieves the re-export * dependencies of es6 modules", function () {
    var deps = detective('export * from "mylib";');
    expect(deps).toMatchSnapshot();
  });

  it("handles multiple imports", function () {
    var deps = detective(
      'import {foo as Foo, bar} from "mylib";\nimport "mylib2"'
    );

    expect(deps).toMatchSnapshot();
  });

  it("handles default imports", function () {
    var deps = detective('import foo from "foo";');

    expect(deps).toMatchSnapshot();
  });

  it("returns an empty list for non-es6 modules", function () {
    var deps = detective('var foo = require("foo");');
    expect(deps).toMatchSnapshot();
  });

  it("does not throw with jsx in a module", function () {
    expect(() => {
      detective('import foo from "foo"; var templ = <jsx />;');
    }).not.toThrow();
  });

  it("does not throw on an async ES7 function", function () {
    expect(() => {
      detective(
        'import foo from "foo"; export default async function bar() {}'
      );
    }).not.toThrow();
  });
});
