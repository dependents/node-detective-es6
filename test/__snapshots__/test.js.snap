// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detective-es6 accepts an ast 1`] = `Array []`;

exports[`detective-es6 handles default imports 1`] = `
Array [
  Object {
    "default": "foo",
    "name": "foo",
  },
]
`;

exports[`detective-es6 handles multiple imports 1`] = `
Array [
  Object {
    "members": Array [
      Object {
        "alias": "Foo",
        "name": "foo",
      },
      Object {
        "alias": "bar",
        "name": "bar",
      },
    ],
    "name": "mylib",
  },
  Object {
    "name": "mylib2",
  },
]
`;

exports[`detective-es6 retrieves the dependencies of es6 modules 1`] = `
Array [
  Object {
    "alias": "All",
    "default": "Abc",
    "name": "mylib",
    "star": true,
  },
]
`;

exports[`detective-es6 retrieves the re-export * dependencies of es6 modules 1`] = `
Array [
  Object {
    "name": "mylib",
  },
]
`;

exports[`detective-es6 retrieves the re-export dependencies alias of es6 modules 1`] = `
Array [
  Object {
    "members": Array [
      Object {
        "alias": "Foo",
        "name": "foo",
      },
    ],
    "name": "mylib",
  },
]
`;

exports[`detective-es6 retrieves the re-export dependencies of es6 modules 1`] = `
Array [
  Object {
    "members": Array [
      Object {
        "alias": "foo",
        "name": "foo",
      },
      Object {
        "alias": "bar",
        "name": "bar",
      },
    ],
    "name": "mylib",
  },
]
`;

exports[`detective-es6 returns an empty list for non-es6 modules 1`] = `Array []`;
